### ISSUE TITLE
Unique '%' Names will only work if their direct Children of the Node

### ISSUE THREAD

**Issue by jcommander** [eumario/GodotSharpExtras#3]
Unfortunately with NodePath, only direct Children of the Nodes with the Unique Name will be found. This kind of destroys the convenience of unique Names as it's meant as a way to not worry about hierarchical relation between Nodes. 

A fix might be calling Roots GetNode function internally when it tries to Get a Node starting with a '%' so they don't have to be children

**Comment by eumario** [comment]
Correct me if I'm wrong, but the '%' Unique Name, is as the name actually states, "Access as Scene Unique Name", meaning, that they only exist within the scene that they are created in, anything outside of this, the Unique Name will not work.

YouTube Video Explaining the Feature: https://youtu.be/HKaPTdZKG8Y?t=231

**Comment by jcommander** [comment]
Sorry it took me so long.
Yeah it's fine that it can be only in the scene/instance it's created in. My point was that only the first 'generation' of children work with this Extension.
Let me try to demonstrate,
consider the following Lists the SceneTree of one Scene:


**Normal Godot:**

- Other Node ✔️
- Node (Script that accesses unique Names📜)
    - Child 1st Gen ✔️
        -  Child 2nd Gen ✔️

I'm pretty sure all of the Nodes would be able to be accessed by a unique Name with a script set like shown above.


**GodotSharpExtras:**

- Other Node ❌
- Node (Script that accesses unique Names📜)
    - Child 1st Gen ✔️
        -  Child 2nd Gen ❌

In my experience with GodotSharpExtras only one of those nodes would be accessible through unique Names.

 I hope this helped visualize the difference in functionalities.

**Comment by eumario** [comment]
Could you possibly post a minimal sample where this is an issue?  Cause as far as I have been able to verify, I have no problems accessing Scene Unique Names, no matter how far in depth the child is.  I have been able to access Scene Unique children as far as 8 grandchildren down without any issues.

**Comment by jcommander** [comment]
I did a few tests and it seems like I have unexpectedly got closer to the root of this issue.
A personal disturbance I had was that I liked the automatic Naming detection when no string for NodePath is specified but especially with Rider, Variable Naming Conventions (e.g. private variables should start with a _ & lowercase) are not really compatible with Node Names.

If the Name of my Node starts with a _, the Node doesn't seem to be found even though the correct Name appears in the Console Output.

Consider the following Scene Tree and Code:
![Capture](https://user-images.githubusercontent.com/7689370/196174045-6ba61764-c009-4a71-a94c-2ee8df61bc34.PNG)

```
public class GodotSharpExtras : Camera2D
{
    [NodePath]
    private ColorRect _colorRect;

    [NodePath("%Label")]
    private Label _label;
    
    public override void _Ready()
    {
        this.OnReady();
        _colorRect.Color = Colors.Orange;
        _label.Text = "Test";
    }
}
```
![Capture](https://user-images.githubusercontent.com/7689370/196174716-223e27d6-bdb6-4ef9-9e84-be8ca3427d83.PNG)
 
Also, I've noticed that the first "Attempt" always seems to be blank.
One fix might be ignoring the underscore _ at the start of a variable name and maybe ignore capitalization if the name started with a underscore so a Variable like _mycolorRect will also search for %MyColorRect.

**Comment by eumario** [comment]
Ok, I see what the problem is.  Sorry for the lateness in reply, I have been busy with other projects.  Currently, the system is setup to search the children from the node that it is currently in.  I'm going to add a check to get the Owner of the node, and try searching from there as well.

**Comment by eumario** [comment]
What version of GodotSharpExtras are you using by chance?  Cause I just tested with 3.5.1 and GodotSharpExtras 0.3.4, and have not had any issues with the example you provided.

**Comment by jcommander** [comment]
It seems I'm not able to replicate the exact Issue on 3.5.1 and 0.3.4 aswell. I was using Godot 3.5 before so I'm not sure why it works now. Unique Names also seem to be accessible just like with regular Godot, which is nice to see.
Thanks for being such cooperative.

But my final remaining Question is: Will you Support _camelCase Names without having to set the Node's Name to be _camelCase? (so a `private ColorRect _myColorRect;` also looks for the Node `myColorRect`/`MyColorRect`)

This would be very useful, especially looking at the fact that 
a) This is the [Naming Convention set out by Godot](https://docs.godotengine.org/en/stable/tutorials/scripting/c_sharp/c_sharp_style_guide.html#naming-conventions)
b) two Scripts may declare the Node as a public/private Field respectively, which, following the Naming Convention would be (`private _myColorRect` and `public MyColorRect`)

**Comment by eumario** [comment]
Latest commit, and soon to be published 0.3.6 fixes the naming convention, and will now add looking for conversion of _myFieldName to MyFieldName.

**Comment by jcommander** [comment]
Great, thanks a lot!
